presto-docs/src/main/sphinx/develop/functions.rst

```diff
@@ -321,3 +321,66 @@ function follows:
   is used when performing a ``GROUP BY`` aggregation, and an implementation
   will be automatically generated for you, if you don't specify a
   ``AccumulatorStateFactory``
+
+
+Advanced Use Cases
+------------------
+
+Raw Block Inputs
+^^^^^^^^^^^^^^^^
+
+Both scalar and aggregation function annotations allow you to define methods
+which operate on native types. In Java, these native types are ``boolean``,
+``Slice``, and ``long``. However, for parameterized implementations or
+parametric types, the standard Java types can't always be used as they aren't
+able to fully represent the input data.
+
+It is possible to define a method handle which can accept *any* types by using
+``@BlockPosition`` in conjunction with the ``@BlockIndex`` parameters. This
+works for both scalar and aggregation function implementations. An example
+signature is provided below. 
+
+.. code-block:: java
+
+    @ScalarFunction("example")
+    public static Block exampleFunction(
+            @BlockPosition @SqlType("array(int)") Block block,
+            @BlockIndex int index) { /* ...implementation */ }
+
+Applying Generic Types with ``@BlockPosition``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Function signatures that use the ``@BlockPosition`` syntax are also able to
+operate over generic types when the function is defined with a
+``@TypeParameter`` annotation. Augment the ``@BlockPosition`` argument with an
+additional ``@SqlType("T")`` annotation to denote that it accepts an argument
+corresponding to the generic type. This works for both scalar and aggregation
+function implementations.
+
+.. code-block:: java
+
+    @ScalarFunction("example")
+    @TypeParameter("T")
+    public static Block exampleFunction(
+            @BlockPosition @SqlType("T") Block block,
+            @BlockIndex int index) { /* ...implementation */ }
+
+
+Retrieving the Generic Type with ``@TypeParameter``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``@TypeParameter`` annotation can also be added at the beginning of a
+function's argument list to allow the implementation to perform type-specific
+logic. Add a ``Type``-typed argument annotated with ``@TypeParameter`` as
+the first argument of the function signature to get access to the ``Type``.
+This works for both scalar and aggregation functions.
```

Comment:
_This works for both scalar and aggregation functions._

No change. This comment is an observation that this sentence is repeated several times in this new doc section, and that this repetition is a good thing. People don't read docs from top of page to the bottom, they find the topic they need, read it, and go on with their day. Topics should be able to stand alone and this repetition allows the topics in _Advanced Use Cases_ to do so.